tests/components/modbus/conftest.py

```diff
@@ -75,27 +84,43 @@ async def mock_modbus(hass, caplog, request, do_config):
                 CONF_HOST: TEST_MODBUS_HOST,
                 CONF_PORT: TEST_PORT_TCP,
                 CONF_NAME: TEST_MODBUS_NAME,
-                **do_config,
+                **conf,
             }
         ]
     }
     mock_pb = mock.MagicMock()
     with mock.patch(
         "homeassistant.components.modbus.modbus.ModbusTcpClient", return_value=mock_pb
     ):
-        mock_pb.read_coils.return_value = ReadResult([0x00])
-        read_result = ReadResult([0x00, 0x00])
-        mock_pb.read_discrete_inputs.return_value = read_result
-        mock_pb.read_input_registers.return_value = read_result
-        mock_pb.read_holding_registers.return_value = read_result
-        if request.param["testLoad"]:
-            assert await async_setup_component(hass, DOMAIN, config) is True
+        if register_words is None:
+            exc = ModbusException("fail read_coils")
```

Comment:
Consider breaking the side effect out to a separate fixture too instead to make it more explicit what happens. We can define default fixtures for both the side effect and the return value.

Reply:
I see what you mean, I will try to make a separate fixture, I am not sure it just should do side effect or also set the values...in both cases it will streamline mock_modbus.

I will do it in a followup PR, it it is alright with you.

I solved the CI error, it was a commit missing.

Reply:
Sure!